CONTRIBUTING.md

```diff
@@ -0,0 +1,245 @@
+# CONTRIBUTING
+
+Information for anyone who would like to contribute to this repository.
+
+- [CONTRIBUTING](#contributing)
+  - [Repository map](#repository-map)
+  - [Generic](#generic)
+    - [Makefile](#makefile)
+    - [pre-commit](#pre-commit)
+  - [Matlab](#matlab)
+    - [Style guide](#style-guide)
+    - [Tests](#tests)
+      - [Adding new tests](#adding-new-tests)
+    - [Continuous integration](#continuous-integration)
+      - [Tests](#tests-1)
+      - [Demos](#demos)
+  - [Python](#python)
+    - [Style guide](#style-guide-1)
+    - [Tests](#tests-2)
+      - [Demos](#demos-1)
+    - [Continuous integration](#continuous-integration-1)
+
+## Repository map
+
+```bash
+├── .git
+├── .github
+│   └── workflows         # Github continuous integration set up
+├── examples              # Python demos
+│   ├── data
+│   ├── example1outputs
+│   └── example2outputs
+├── glmsingle             # Python implementation
+│   ├── cod
+│   ├── design
+│   ├── gmm
+│   ├── hrf
+│   ├── ols
+│   ├── ssq
+│   └── utils
+├── matlab                # Matlab implementation
+│   ├── examples          # Matlab demos
+│   ├── fracridge         # Fracridge submodule
+│   └── utilities
+└── tests                 # Python and Matlab tests
+    ├── data              # Data used as inputs for the tests
+    └── expected          # Expected results of the tests
+│       └── matlab
+
+```
+
+## Generic
+
+### Makefile
+
+A `Makefile` is used to help set / and automate some things.
+
+In a terminal type `make help` to see what some of the different "recipes" you
+can run with this `Makefile`.
+
+See
+[here for a short intro on using `Makefiles`](https://the-turing-way.netlify.app/reproducible-research/make.html)
+
+### pre-commit
+
+You can use the [`pre-commit` python package](https://pre-commit.com/) in this
+repo to make sure you only commit properly formatted files (for example `.yml`
+files).
+
+1. Install `pre-commit`
+
+```bash
+$ pip3 install pre-commit
+```
+
+It is also included in `requirements_dev.txt`, so it will installed by running:
+
+```bash
+$ pip3 install -r requirements_dev.txt
+```
+
+The `.pre-commit-config.yml` file defines the checks to run when committing
+files.
+
+1. Run the following command to install the `pre-commit` "hooks"
+
+```bash
+$ pre-commit install
+```
```

Comment:
`.yml` files have a pretty strict syntax so using pre-commit can of great help to make sure you do not commit things YML files that can't be parsed.